From 0aea3e2629cd1681d8d8e6e0a4409959b31ea4e9 Mon Sep 17 00:00:00 2001
From: Anil Kumar Ch <anilkumar@ti.com>
Date: Thu, 10 Nov 2011 15:21:47 +0100
Subject: [PATCH 7/7] can: d_can: am335x profile modification for dcan0

This patch modifies the profile information of am335x device.

Profile reads of cpld_client from smbus gives an error leads to put
the device into default profile 0. So by default the board configured
to beaglebone even if we set the sw8 switch on daughter card to other
profiles.

This patch makes all the IO connecters are configured in profile 1.

Signed-off-by: Anil Kumar Ch <anilkumar@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 590c4ca..ac64fc7 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -340,6 +340,9 @@ static u32 am335x_get_profile_selection(void)
 {
 	int val = 0;
 
+	/* FIXME: temporary fix */
+	return 1;
+
 	if (!cpld_client)
 		/* error checking is not done in func's calling this routine.
 		so return profile 0 on error */
@@ -1721,8 +1724,8 @@ out:
 	 */
 	pr_err("Could not detect any board, falling back to: "
 		"Beaglebone (< Rev A3) with no daughter card connected\n");
-	daughter_brd_detected = false;
-	setup_beaglebone_old();
+	daughter_brd_detected = true;
+	setup_general_purpose_evm();
 
 	/* Initialize cpsw after board detection is completed as board
 	 * information is required for configuring phy address and hence
-- 
1.7.2.5

